chore: Fix tests, remove EOL OSes, fix Ruby 3.4 compatibility warnings

RulerOf opened the pull request:
# Description

Most of the tests were broken.

## Issues Resolved

I didn't create any, but this will fix the broken tests on #420 

## Type of Change

chore

## Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] Commit message includes a [Conventional Commit Message](https://www.conventionalcommits.org/en/v1.0.0)

Stromweld committed:
fix: release please configs

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>
Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

add codeowners

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

RulerOf committed:
fix: Support docker build output for Docker Desktop v4.31

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: remove unused integration test files

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
feat: Ruby 3.4.0 compatibility

Ruby is removing some gems from the standard library. Require them explicitly.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
feat: Use slightly-newer inspec in development

Development group gems resolved an old version of inspec on my machine, and would throw an ugly warning. Bump the minimum inspec version to resolve that.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
feat: Test dockerfile build with Oracle Linux

CentOS 7 is EOL. Stop testing it, and switch to a similar release.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
feat: Stop installing Chef in every test suite

Chef is not supported on every OS that kitchen-docker supports, so it doesn't really make sense to install it here. Also Dokken is a thing.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: Switch all tests to inspec verifier

The default busser verifier literally refuses to work, and the 1-test inspec is simple enough to work on everything.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: Update gitignore

Bundler shims and rubymine folder

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: Remove EOL operating systems from test suite

CentOS 7, Ubuntu 18, Rocky 8, Debian 11 are all EOL.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: Fix yaml formatting

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: More restrictive Inspec pin

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

tas50 commented on kitchen.yml:
```diff
@@ -4,31 +4,44 @@
+      image: opensuse/leap:15.5
+  - name: opensuse-15.6
     driver:
-      image: opensuse/leap:15
```
This slug should get you the latest without having to maintain the version here

RulerOf replied:
I ended up putting in the two versions separately because both of them showed to still be under active support on the website.

RulerOf replied:
Oh wait I see what you mean. Specify a "latest v15" in addition to a "specific v15."

tas50 replied:
Leap:15 should always be the latest

RulerOf committed:
chore: remove inspec suite

Everything is already testing with Inspec

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

Stromweld commented on test/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM centos:7
+FROM oraclelinux:7
```
in the kitchen-dokken driver we switched to almalinux-9. Should probably do the same here for consistency and future proofing. Then the yum commands can be updated to dnf.

RulerOf replied:
I updated the Dockerfile and got it building successfully, but kitchen-docker complains the container isn't listening for ssh connections.

Perhaps I need to change the CMD to run the ssh server? It wasn't obvious to me why it worked with Oracle 7 but not Alma 9.

RulerOf committed:
fix: re-require lib/docker/version.rb

The verifier doesn't work without it. Not sure why it was removed, but I know base PR was ongoing.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: Rework test suites to use kitchen.yml

Add a step to the github action that uses an embedded ruby script to discover the list of suites directly from the kitchen config.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
chore: test supported opensuse 15.5 and latest 15

Do this instead of testing 15.5 and 15.6 explicitly

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
feat: Implement `kitchen login` for docker transport

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>

RulerOf committed:
fix: Use newer syntax for ENV variables

Current versions of Docker Desktop will output a LegacyKeyValueFormat
warning when using 'ENV key value' syntax. We switch to the updated 'ENV
key=value' syntax.

Signed-off-by: Andrew Bobulsky <rulerof@gmail.com>
